tiledb/ml/readers/pytorch_sparse.py

```diff
@@ -111,8 +112,9 @@ def __iter__(self) -> Iterator[DataType]:
         # Loop over batches
         for offset in range(iter_start, iter_end, self.batch_size):
             # Yield the next training batch
-            x_batch = self.x[offset : offset + self.batch_size]
-            y_batch = self.y[offset : offset + self.batch_size]
+            x_batch, y_batch = run_io_tasks_in_parallel(
```

Comment:
Just noticed that this is called in a loop; creating and destroying a `ThreadPoolExecutor` on every loop is not cheap/free. I'd move the `run_io_tasks_in_parallel` in this class and change the executor to an attribute created once in `__init__` (or make it a mixin class and reuse it since it's used in more than one classes).

Reply:
Good point I will change that!